test/formats/test_apev2.py

```diff
@@ -87,7 +87,7 @@ def test_supports_tags(self):
             supports_tag = self.format.supports_tag
             for key in VALID_KEYS + SUPPORTED_TAGS:
                 self.assertTrue(supports_tag(key), '%r should be supported' % key)
-            for key in INVALID_KEYS + apev2.UNSUPPORTED_TAGS:
+            for key in INVALID_KEYS + list(apev2.UNSUPPORTED_TAGS):
```

Comment:
Here we can go a bit further by also converting `VALID_KEYS` and `INVALID_KEYS` to sets:

```python
VALID_KEYS = {
    ' valid Key}',
    '{ $ome tag~}',
    'xx',
    'x' * 255,
}

INVALID_KEYS = {
    'invalid\x7fkey',
    'invalid\x19key',
    '',
    'x',
    'x' * 256,
    'ID3',
    'TAG',
    'OggS',
    'MP+',
}


SUPPORTED_TAGS = set(TAGS) - set(apev2.UNSUPPORTED_TAGS)


class CommonApeTests:

    class ApeTestCase(CommonTests.TagFormatsTestCase):
        def setup_tags(self):
            super().setup_tags()
            self.unsupported_tags['r128_album_gain'] = '-2857'
            self.unsupported_tags['r128_track_gain'] = '-2857'

        def test_supports_tags(self):
            supports_tag = self.format.supports_tag
            for key in VALID_KEYS | SUPPORTED_TAGS:
                self.assertTrue(supports_tag(key), '%r should be supported' % key)
            for key in INVALID_KEYS | apev2.UNSUPPORTED_TAGS:
                self.assertFalse(supports_tag(key), '%r should be unsupported' % key)
```

Note the use of `|` operator instead of `+`

Also, since `set(TAGS)` is equivalent to `set(TAGS.keys()`

Reply:
Indeed, that simplifies code even a bit and makes it more readable. Did the changes in https://github.com/metabrainz/picard/pull/1935/commits/aed015c6570cebba5f0a69b12962bd250e1fc630